permission_prompt/explainer.md

```diff
@@ -0,0 +1,151 @@
+
+Long story short, there does not seem to be a way to get out of this bind and achieve our goals with WebTransport.
+
+## A proposal
+
+The previous attempts motivated the switch to a permission-based alternative. We propose that secure contexts be allowed to fetch resources from private network servers over plaintext HTTP - circumventing mixed content restrictions - given express user permission.
+
+At a high level, websites indicate their intention to access a private network server using some new API. Chrome queries the server for some identification information it can display to the user in a permission prompt. The user then chooses if they desire to allow the website to access the target server.
+
+This would mean that websites that wish to speak to private network devices must be served over HTTPS. The target device, however, would not have to serve HTTPS. It would only need to respond correctly to CORS preflights, and maybe to some kind of simple identification request.
+
+### Triggering
+
+Some kind of API to trigger the permission request when accessing a private network server for the first time is required. Here we propose adding a new parameter to the `fetch()` options bag. For example:
+
+```
+fetch("http://router.private/ping", {
+  targetAddressSpace: "private",
+});
+```
+
+This would instruct the browser to allow the fetch even though the scheme is non-secure and obtain a connection to the target server. This ensures the feature cannot be abused to bypass mixed content in general.
+
+If the remote IP address does not belong to the IP address space specified as the privateNetworkAccess option value, then the request is failed.
+
+### Preflighting 
+
+If it does belong, then a CORS preflight request is sent as specified today. The target server then responds with a CORS preflight response, augmented with the following two headers:
+
+```
+Private-Network-Access-Name: <some human-readable device self-identification>
+Private-Network-Access-ID: <some unique and stable machine-readable ID, such as a MAC address>
+```
+
+For example:
+
+```
+Private-Network-Access-Name: "My Smart Toothbrush"
+Private-Network-Access-ID: "01:23:45:67:89:0A"
+```
+
+### Gaining permission
+
+A prompt is then shown to the user asking for permission to access the target device. Something like:
+
+![prompt](prompt.png)
+
+The `-Name` header is used to present a friendly string to the user instead of, or in addition to, an origin (often a raw IP literal).
+
+The `-ID` header is used to key the permission and recognize the device across IP addresses. Indeed, widespread use of DHCP means that devices are likely to change IP addresses regularly, and we would like to avoid both cross-device confusion and permission fatigue.
```

Comment:
Shall we mention that this is less certain? We might or might not want to do this, IIRC.